### PR TITLE
Adds Highlander Style: The Martial Art of Highlanders

### PR DESCRIPTION
**Pull request by FalseIncarnate**
Adds a new martial art: Highlander Style!
- Wielders of the powerful Highlander claymores are imbued with this mythical prowress. Should your both your hands ever lack a grip on a Highlander claymore, you will relinquish your newfound skill.
- Highlander Style grants immunity to ranged attacks and an unwavering distain of cowardly guns. Just like with Sleeping Carp, you will deflect projectiles and cannot fire guns (but can pick them up to throw in the trash where they belong).

Adds new weapon: Highlander Claymore
- Grants the wielder the Highlander Style while held in either hand.
- Normal claymores were left in, but do not provide the martial art (so they are "safe" to use outside highlander)

Minor refactor to martial arts which prevent use of guns.
- No longer relies on the name of the style, but rather the new aptly-named var: `can_use_guns`

:cl:
rscadd: Adds Highlander Style, granted to the wielder of Highlander Claymores. This martial art allows you to deflect ranged attacks from the weapons of COWARDS. FOR THE HONOR OF THE HIGHLANDERS!
tweak: Highlander now equips combatants with a Highlander claymore instead of a normal claymore. FIGHT ON BROTHERS!
/:cl:

### CODE REVIEW

**Comment by Bram961** [comment]
YES!

**Comment by Anasari** [comment]
Finally!

**Comment by KingPhilipIII** [comment]
Scotland forever brothers!

**Comment by taukausanake** [comment]
I still want a martial art that let's you catch projectiles of any types and throw them back. This however does help out with the cheesy-ness of Highlander.

Does this require the claymore to be wielded or in either hand? The OP doesn't make that clear

**Comment by FalseIncarnate** [comment]
@taukausanake Claymores (despite what you would think) are not two-handed weapons, and thus cannot be wielded in two hands like a chainsaw or staff. You gain the Highlander Style as long as at least one of your hands contains a Highlander Claymore. This means you lose the effect if you put the claymore on your belt or drop it if you don't have one in your other hand.

File: code/game/objects/items/weapons/highlander_swords.dm [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+			style.teach(H, 1)
+			to_chat(H, "<span class='notice'>THERE CAN ONLY BE ONE!</span>")
+	else if(H.martial_art && H.martial_art == style)
+		style.remove(H)
+		var/obj/item/weapon/claymore/highlander/sword = H.is_in_hands(/obj/item/weapon/claymore/highlander)
+		if(sword)
+			//if we have a highlander sword in the other hand, relearn the style from that sword.
+			sword.style.teach(H, 1)
+
+	return
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/highlander_swords.dm]: Trailing return.

File: code/game/objects/items/weapons/highlander_swords.dm [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+/obj/item/weapon/claymore/highlander/dropped(mob/user)
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/H = user
+	style.remove(H)
+	var/obj/item/weapon/claymore/highlander/sword = H.is_in_hands(/obj/item/weapon/claymore/highlander)
+	if(sword)
+		//if we have a highlander sword in the other hand, relearn the style from that sword.
+		sword.style.teach(H, 1)
+	return
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/highlander_swords.dm]: Trailing return.

File: code/game/objects/items/weapons/highlander_swords.dm [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	no_guns_message = "You'd never stoop so low as to use the weapon of a COWARD!"
+
+
+//Highlander Claymore
+//	Grants the wielder the Highlander Style Martial Art
+
+/obj/item/weapon/claymore/highlander
+	name = "Highlander Claymore"
+	desc = "Imbues the wielder with legendary martial prowress and a nigh-unquenchable thirst for glorious battle!"
+	var/datum/martial_art/highlander/style = new
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/highlander_swords.dm]: Get rid of this datum in a Destroy

Comment (FalseIncarnate) [inline code/game/objects/items/weapons/highlander_swords.dm]: woops, forgot about that.

File: code/modules/martial_arts/martial.dm [hunk 4]
```diff
@@ -7,6 +7,8 @@
 	var/datum/martial_art/base = null // The permanent style
 	var/deflection_chance = 0 //Chance to deflect projectiles
 	var/help_verb = null
+	var/can_use_guns = 1	//set to zero to prevent users of this style from using guns (sleeping carp, highlander). They can still pick them up, but not fire them.
```
Comment (Fox-McCloud) [inline code/modules/martial_arts/martial.dm]: As this entire system is from TG, and as they have the same functionality, having the same var names is beneficial to both of us, in the long run.

Please make this the same as theirs: `var/no_guns = FALSE`

**Comment by Fox-McCloud** [comment]
Overall looks good; just one last thing.

**Pull request merged**